src/game/server/gamecontext.cpp

```diff
@@ -129,28 +129,109 @@ class CCharacter *CGameContext::GetPlayerChar(int ClientID)
 	return m_apPlayers[ClientID]->GetCharacter();
 }
 
-int CGameContext::GetZombieCount() {
-	int count = 0;
+int CGameContext::GetActivePlayerCount(){
+	
+	// returns how many players are currently playing and not spectating
+		int PlayerCount = 0;
+		int SpecCount = 0;
+		for(int i=0; i<MAX_CLIENTS; i++)
+		{
+			//COMPILER ERROR ‘class IServer’ has no member named ‘m_aClients’
+			if(Server()->m_aClients[i].m_State == CServer::CClient::STATE_INGAME)
+			{
+				if (m_apPlayers[i]->GetTeam() != TEAM_SPECTATORS)
+					PlayerCount++;
+				else 
+					SpecCount++;
+			}
+		}
+		
+		m_NbActivePlayers = PlayerCount;
+		m_NbSpectators = SpecCount;
+
+		dbg_msg("infclass", "Active Players: %d -- Spectators: %d", m_NbActivePlayers, m_NbSpectators);
+		
+		return PlayerCount;
+}
+
+int CGameContext::GetSpectatorCount(){
+	
+	int PlayerCount = 0;
+	int SpecCount = 0;
+	for(int i=0; i<MAX_CLIENTS; i++)
+	{
+		
+		//COMPILER ERROR ‘class IServer’ has no member named ‘m_aClients’
+		if(Server()->m_aClients[i].m_State == CServer::CClient::STATE_INGAME && m_apPlayers[i]->GetTeam() == TEAM_SPECTATORS)
```

Comment:
> class CServer : public IServer

CServer inherits from IServer
CServer has the m_aClients array but IServer does not have them
I dont know jet how to get the CServer object